File: 092_Reverse_Linked_List_II/092_1.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseBetween(struct ListNode* head, int m, int n){
    int i = 0;
    struct ListNode *dummy = (struct ListNode *)malloc(sizeof(struct ListNode));
    dummy->val = -1;
    dummy->next = head;
    struct ListNode *front = dummy;
    while (--m) {
        front = front->next;
        i++;
    }
    struct ListNode *prev = NULL;
    head = front->next;

    while (i < n) {
        struct ListNode *temp = head->next;
        head->next = prev;
        prev = head;
        head = temp;
        i++;
    }
    front->next->next = head;
    front->next = prev;
    return dummy->next;
}

File: 092_Reverse_Linked_List_II/092_1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseBetween(self, head: ListNode, m: int, n: int) -> ListNode:
        dummy = ListNode(-1)
        dummy.next = head
        pre = dummy
        for _ in range(m - 1):
            pre = pre.next
        head = pre.next
        for i in range(m, n):
            nxt = head.next
            head.next = nxt.next
            nxt.next = pre.next
            pre.next = nxt
        return dummy.next